### PR TITLE
Handle QIs not giving back a 404

### PR DESCRIPTION
**Pull request by orta**
We were getting a 400 ( probably ) from the QI server while it was being hammered, see #234 - so now we glob a range of 40X replies and give a 404.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -143,7 +143,7 @@ def quality_indicator_group input
     @quality, response_code = PodQualityEstimate.load_quality_estimate(@name)
     @page_title = "#{@name}'s Quality Estimate on CocoaPods.org"
 
-    if response_code == 404
+    if (400..404).include? response_code
```
Comment (segiddins) [inline app.rb]: Why not 400...499?

Comment (orta) [inline app.rb]: I wasn't sure, which is why I left the one open, I'm happy to do that instead

**Comment by segiddins** [comment]
@orta let's :ship: this?

**Comment by orta** [comment]
Yeah!

**Comment by floere** [comment]
Did we ever confirm there was a hammering? I don't believe so.

File: app.rb [hunk 2]
```diff
@@ -143,7 +143,7 @@ def quality_indicator_group input
     @quality, response_code = PodQualityEstimate.load_quality_estimate(@name)
     @page_title = "#{@name}'s Quality Estimate on CocoaPods.org"
 
-    if response_code == 404
+    if (400...600).cover? response_code
       not_found
```
Comment (kylef) [inline app.rb]: Should we really be returning a not found error when we don't actually know if it's not found?

**Pull request merged**